test/CodeGenerator/AbstractInjectorTest.php

```diff
@@ -187,17 +187,9 @@ public function testConstructionWithoutContainerUsesDefaultContainer()
     public function testFactoryIsCreatedFromClassNameString()
     {
         $subject = $this->createTestSubject(function () {
-            return ['SomeClass' => StdClassFactory::class ];
+            return ['SomeClass' => StdClassFactory::class];
         });
 
-        $this->assertSame(
-            StdClassFactory::class,
-            self::readAttribute($subject, 'factories')['SomeClass'] ?? null
-        );
         $this->assertInstanceOf(stdClass::class, $subject->create('SomeClass'));
-        $this->assertInstanceOf(
-            StdClassFactory::class,
-            self::readAttribute($subject, 'factoryInstances')['SomeClass'] ?? null
-        );
```

Comment:
what is the reason of removing these assertions?

I know that probably these are not the best, as these checks attributes and we really shouldn't do that.

Reply:
See my Review comment. The test should not check the internal state and it will stop working with PHPUnit 9

Reply:
Yes, I agree it should not. The problem is that there was no any other way to check it, that's why it was done that way (and we have it in many components....).

Reply:
I know, and it was me who added them. We could use Reflection directly, but it still feels kinda hacky and bad.
The question is: do we really need this kind of assertion, or could we do better with infection and benchmark testing here (two tools I'd like to be added anyways).

What do we want to test here? A factory class name should be accepted or more? If more: does it really add value?

That's some topic where I'd like to hear @ocramius opinion, if he could spend a minute

Reply:
@michalbundyra I re-added the assertions with reflection, but I still don't think that they add any real value to the test.